src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...actions[i].extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        const keysToRemove = actions[i].keysToRemove;
+        const { ...newState } = currentState;
+
+        keysToRemove.forEach(key => delete newState[key]);
+        currentState = newState;
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
```

Comment:
![image](https://user-images.githubusercontent.com/96409563/223962542-e8c1b904-68ae-4990-8229-9148b7127ed3.png)

Reply:
still not fixed. please add the default case for switch/case.